src/content/plugins/module-federation-plugin.md

```diff
@@ -0,0 +1,211 @@
+
+This syntax provides you more control over each shared library in which you can define package name as the key and version ([semver](https://semver.org/)) as the value.
+
+```js
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      shared: {
+        // adds lodash as shared module
+        // version is inferred from package.json
+        // it will use the highest moment version that is >= 4.17 and < 5
+        lodash: '^4.17.0',
+      },
+    }),
+  ],
+};
+```
+
+#### Object syntax with sharing hints
+
+This syntax allows you to provide additional [hints](#sharing-hints) to each shared package where you define the package name as the key, and the value as an object containing hints to modify sharing behavior.
+
+```js
+const deps = require('./package.json').dependencies;
+
+module.exports = {
+  plugins: [
+    new ModuleFederationPlugin({
+      shared: {
+        // adds react as shared module
+        // version is inferred from package.json
+        // it will always use the shared version, but print a warning when the shared lodash is < 4.17 or >= 5
+        react: {
+          requiredVersion: deps.lodash,
+          singleton: true,
+        },
+      },
+    }),
+  ],
+};
+```
+
+### Sharing hints
+
+#### **`eager`**
+
+`boolean`
+
+This hint will allow Webpack to include the provided and fallback module directly instead of fetching the library via an asynchronous request. In other words, this allows to use this shared module in the initial chunk. Also, be careful that all provided and fallback modules will always be downloaded when this hint is enabled.
+
+#### **`import`**
+
+`false | string`
+
+The provided module that should be placed in the shared scope. This provided module also acts as fallback module if no shared module is found in the shared scope or version isn't valid. (The value for this hint defaults to the property name.)
+
+#### **`packageName`**
+
+`string`
+
+The package name that is used to determine required version from description file. This is only needed when the package name can't be automatically determined from request.
+
+#### **`requiredVersion`**
+
+`false | string`
+
+The required version of the package.
+
+#### **`shareKey`**
+
+`string`
+
+The requested shared module is looked up under this key from the shared scope.
+
+#### **`shareScope`**
+
+`string`
+
+The name of the shared scope.
+
+#### **`singleton`**
+
+`boolean`
+
+This hint only allows a single version of the shared module in the shared scope (disabled by default). Some libraries use a global internal state (e.g. react, react-dom). Thus, it is critical to have only one instance of the library running at a time.
+
+#### **`strictVersion`**
+
+`boolean`
+
+This hint allows Webpack to reject the shared module if version is not valid (defaults to `true`, if local fallback module is available and shared module is not a singleton, otherwise `false`, it has no effect if there is no required version specified).
```

Comment:
```suggestion
This hint allows webpack to reject the shared module if version is not valid (defaults to `true` when local fallback module is available and shared module is not a singleton, otherwise `false`, it has no effect if there is no required version specified).
```